xbmc/network/DNSNameCache.cpp

```diff
@@ -106,6 +112,13 @@ bool CDNSNameCache::Lookup(const std::string& strHostName, std::string& strIpAdd
 
 bool CDNSNameCache::GetCached(const std::string& strHostName, std::string& strIpAddress)
 {
+#if !defined(TARGET_WINDOWS) && defined(HAS_FILESYSTEM_SMB)
+  WSDiscovery::CWSDiscoveryPosix& WSInstance =
+      dynamic_cast<WSDiscovery::CWSDiscoveryPosix&>(CServiceBroker::GetWSDiscovery());
+  if (WSInstance.GetCached(strHostName, strIpAddress))
+    return true;
+#endif
+
```

Comment:
This will override cache entries added via <hosts> in advanced settings, so it should be moved further down

Reply:
Corrected.